### PR TITLE
[FIXED JENKINS-31596] Provide an extension point for MasterKillSwitchConfiguration relevance

### PR DESCRIPTION
**Pull request by stephenc**
@reviewbybees

### CODE REVIEW

File: core/src/main/java/jenkins/security/s2m/MasterKillSwitchConfiguration.java [hunk 1]
```diff
@@ -52,8 +54,39 @@ public boolean isRelevant() {
...
+     * by implementing this extension point they can indicate such additional conditions.
+     *
+     * @since FIXME
+     */
+    public static abstract class Relevance implements ExtensionPoint {
+
+        /**
+         * Is the {@link MasterKillSwitchConfiguration} relevant.
+         *
+         * @return {@code true} if the {@link MasterKillSwitchConfiguration} relevant.
```
Comment (jtnord) [inline core/src/main/java/jenkins/security/s2m/MasterKillSwitchConfiguration.java]: :nit:  is relevant.

**Comment by jtnord** [comment]
:bee:

**Comment by varmenise** [comment]
:bee:

**Comment by jglick** [comment]
Sounds useful but where is the matching PR to `swarm-plugin` showing the extension point implementation?

**Comment by stephenc** [comment]
I am awaiting the core release. Implementation will just be

```
@Extension
public class RelevanceImpl extends MasterKillSwitchConfiguration.Relevance {
  public boolean isRelevant() {
    return true; // bonus points if checking that there is a swarm configured first
  }
}
```

**Comment by jglick** [comment]
> I am awaiting the core release.

For the benefit of reviewers it is helpful to have that PR in place even using a snapshot dependency.

**Comment by jglick** [comment]
BTW given the ever-expanding list of conditions, I have to ask why we do not simply delete `isRelevant` and unconditionally show the switch. The help text could explain that if you never use slaves then you do not need this defense. Of course if you never use slaves, you effectively have no security anyway, so what are you doing in `/configureSecurity` to begin with? I think we are trying to be too clever.

**Comment by jglick** [comment]
@stephenc could this or something related have caused [this failure](https://jenkins.ci.cloudbees.com/job/core/job/acceptance-test-harness/lastSuccessfulBuild/testReport/plugins/MatrixAuthPluginTest/helloWorld/)?

```
hudson.security.AccessDeniedException2: anonymous is missing the Overall/RunScripts permission
 at hudson.security.ACL.checkPermission(ACL.java:63)
 at hudson.model.Node.checkPermission(Node.java:462)
 at jenkins.security.s2m.AdminWhitelistRule.setMasterKillSwitch(AdminWhitelistRule.java:210)
 at jenkins.security.s2m.MasterKillSwitchConfiguration.configure(MasterKillSwitchConfiguration.java:41)
 at hudson.security.GlobalSecurityConfiguration.configureDescriptor(GlobalSecurityConfiguration.java:126)
 at hudson.security.GlobalSecurityConfiguration.configure(GlobalSecurityConfiguration.java:115)
 at hudson.security.GlobalSecurityConfiguration.doConfigure(GlobalSecurityConfiguration.java:79)
```

**Comment by stephenc** [comment]
Argh!

**Comment by stephenc** [comment]
https://github.com/jenkinsci/jenkins/pull/1929 is the follow-up @jglick

**Pull request merged**